### DISCUSSION TITLE
Lack of animation in TabView + ForEachStore

### DISCUSSION THREAD

**Discussion by Alex0Klain** [pointfreeco/swift-composable-architecture#2166] [Q&A] [answered]
Hello 👋🏻
I use a bundle from `TabView` (with `page` style) and inside `ForEachStore` with screens. When the `index` changes, the `TabView` does not animate the screen change. Although if you just leave the `TabView` without the `ForEachStore`, then the animation will be. Is there anything we can do about it?

**Code example:**
```
@ViewBuilder
fileprivate var pages: some View {
  WithViewStore(
    self.store.scope(state: { $0.currentIndex })
  ) { viewStore in
    TabView(
      selection: viewStore.binding(
        get: { $0 },
        send: TVMovieDetailsPagerActions.updateCurrentMovieIndex
      )
    ) {
      ForEachStore(
        self.store.scope(
          state: \.moviePages,
          action: TVMovieDetailsPagerActions.moviePage(id:action:)
        )
      ) { movieStore in
        TVMovieDetailsView(movieStore)
      }
    }
    .tabViewStyle(.page(indexDisplayMode: .never))
    .animation(.spring(), value: viewStore.state)
    .transition(.slide)
  }
}
```

**Comment by stephencelis** [comment]
@Alex0Klain SwiftUI's animations can be tricky even in a vanilla project, and the `animation` and `transition` modifiers don't always work the way I'd expect, so I'm wondering if you've reproduced the issue outside of TCA?

Without a working code sample it's hard to troubleshoot further, but one thing you could try is tacking on an `animation` modifier directly to the binding:

```diff
 TabView(
   selection: viewStore
     .binding(
       get: { $0 },
       send: TVMovieDetailsPagerActions.updateCurrentMovieIndex
     )
+    .animation(.spring())
 ) {
```

**Reply by Alex0Klain** [reply]
I just tried adding a modifier to `Binding`, but unfortunately it didn't help. From what I've noticed, the problem is only when using the `ForEachStore`. Even if you add a regular `view` with `text` to the `ForEachStore`, the problem is reproduced. With a normal `TabView` + `ForEach` animation works :\

**Reply by stephencelis** [reply]
@Alex0Klain Any chance you can put together a working demo of the problem? We'd love to dive in and see if we can find the problem.

**Reply by Alex0Klain** [reply, marked as answer]
I wrote a sample project. And that's where it works. It looks like the problem is something else. I will figure it out, thank you for responding 🤔
https://pastebin.com/mUwsjBBq
